denguefever_tw/dengue_linebot/denguebot_fsm/denguebotfsm.py

```diff
@@ -213,6 +221,44 @@ def _create_hospitals_msgs(self, hospital_list):
         hospital_messages = [template_message]
         return hospital_messages
 
+    def _create_zapper_imgmap(self, event):
+        zapper_imgmap = ImagemapSendMessage(
+            base_url='https://i.imgur.com/9piGQjS.jpg',
+            alt_text='user zapper information',
+            base_size=BaseSize(height=1040, width=1040),
+            actions=[
+                URIImagemapAction(
+                    link_uri='',
+                    area=ImagemapArea(
+                        x=0, y=0, width=520, height=520
+                    )
+                ),
+                MessageImagemapAction(
+                    text='我想了解整個商圈的蚊蟲情況',
+                    area=ImagemapArea(
+                        x=520, y=0, width=520, height=520
+                    )
+                ),
+                MessageImagemapAction(
+                    text='我的補蚊燈需要專人協助',
+                    area=ImagemapArea(
+                        x=0, y=520, width=520, height=520
+                    )
+                ),
+                MessageImagemapAction(
+                    text='我要綁定補蚊燈！',
+                    area=ImagemapArea(
+                        x=520, y=520, width=520, height=520
+                    )
+                ),
+            ]
+        )
+
+        line_user = LineUser.objects.get(user_id=event.source.user_id)
+        if line_user.zapper_id:
+            zapper_imgmap.actions[0].link_uri = 'https://example.com/' + line_user.zapper_id
```

Comment:
It would be better if you use `format` to concatenate strings.
Use `+` to concatenate strings is slow in Python.

Reply:
Fixed